src/libstore-c/nix_api_store.cc

```diff
@@ -216,4 +216,94 @@ nix_err nix_store_copy_closure(nix_c_context * context, Store * srcStore, Store
     NIXC_CATCH_ERRS
 }
 
+nix_err nix_store_drv_from_path(
+    nix_c_context * context,
+    Store * store,
+    const StorePath * path,
+    void (*callback)(void * userdata, const nix_derivation * drv),
+    void * userdata)
+{
+    if (context)
+        context->last_err_code = NIX_OK;
+    try {
+        nix::Derivation drv = store->ptr->derivationFromPath(path->path);
+        if (callback) {
+            const nix_derivation tmp{drv};
+            callback(userdata, &tmp);
+        }
+    }
+    NIXC_CATCH_ERRS
+}
+
+nix_err nix_derivation_get_outputs(
+    nix_c_context * context,
+    const nix_derivation * drv,
+    void (*callback)(void * userdata, const char * name, const nix_derivation_output * drv_output),
+    void * userdata)
+{
+    if (context)
+        context->last_err_code = NIX_OK;
+    try {
+        if (callback) {
+            for (const auto & [name, result] : drv->drv.outputs) {
+                const nix_derivation_output tmp{result};
+                callback(userdata, name.c_str(), &tmp);
+            }
+        }
+    }
+    NIXC_CATCH_ERRS
+}
+
+nix_err nix_derivation_get_outputs_and_optpaths(
```

Comment:
This should just be the opt paths, which is useful for the FOD case. The outputs themselves can be gotten from the JSON.

Reply:
Why?

Reply:
This is me elaborating what I wrote in https://github.com/NixOS/nix/pull/14031#issuecomment-3314139031